1-Draft/RFC00XX-Native-Command-Error-Handling.md

```diff
@@ -0,0 +1,229 @@
+ [implementation plan](https://github.com/PowerShell/PowerShell-RFC/pull/88#issuecomment-613653678)
+
+## Specification
+
+This RFC proposes including native commands in the error handling framework when the feature is
+enabled by adding an error to the error stream if the exit code of a native command is non-zero.
+
+The `$PSNativeCommandErrorAction` preference variable will implement a version of the
+`$ErrorActionPreference` variable for native commands.
+
+- The value will default to `Ignore` for compatibility with existing behavior.
+- For non-zero exit codes and except for the value 'Ignore', an `ErrorRecord` will be added to
+  `$Error` that wraps the exit code and the command executed that returned the exit code.
+- Initially, only the existing values of `$ErrorActionPreference` will be supported.
+- The set of values may be extended later to include `MatchErrorActionPreference`, which should
+  apply the `$ErrorActionPreference` setting to native commands.
+- A conversion will occur between `$PSNativeCommandErrorAction` and `$ErrorActionPreference`
+  values, where `MatchErrorActionPreference` is converted to the current value of
+  `$ErrorActionPreference`
+
+Valid values for `$ErrorActionPreference`
+
+| Value           | Definition
+----------------  | -------------------
+| Break           | Enter the debugger when an error occurs or when an exception is raised.
+| Continue        | (Default) - Displays the error message and continues executing.
+| Ignore          | Suppresses the error message and continues to execute the command. The Ignore value is intended for per-command use, not for use as saved preference. Ignore isn't a valid value for the $ErrorActionPreference variable.
+| Inquire         | Displays the error message and asks you whether you want to continue.
+| SilentlyContinue| No effect. The error message isn't displayed and execution continues without interruption.
+| Stop            | Displays the error message and stops executing. In addition to the error generated, the Stop value generates an ActionPreferenceStopException object to the error stream. stream
+| Suspend         | Automatically suspends a workflow job to allow for further investigation.
+
+The reported error record should be created with the following details:
+
+| Property        | Definition
+----------------  | -------------------
+| exception:      | `ExitCode`, with the exit code of the failed command.
+| error id:       | `"Program {0} failed with unhandled exit code {1}"`, with the command name and the exit code, from resource string `ProgramFailedToComplete`.
+| error category: | `ErrorCategory.NotSpecified`.
+| object:         | exit code
+
+ This does not provide the actual semantics of bash `set -eo pipefail` as Bourne shell-style
+ integration with existing status handling syntax is not implemented.
+
+`Set -eo pipefail` is a combination of `Set -u` and `Set -o pipefail`:
```

Comment:
i think this is platform/shell specific - on my mac at least with bash, `set -eo pipefail` does not also `set -u` as the following transcript shows:
```powershell
PS> bash

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
bash-3.2$ set -eo pipefail
bash-3.2$ echo $SFDKLSF

bash-3.2$ set -u
bash-3.2$ echo $SFDKLSF
bash: SFDKLSF: unbound variable
PS>
```

Reply:
In light of the addition of set -o pipefail to the rfc, I cleaned up the definitions in the motivation section and removed set -eo.